crates/iota-light-client/src/utils.rs

```
@@ -319,138 +519,47 @@ pub async fn get_verified_effects_and_events(
         // Since we did not find a small committee checkpoint we use the genesis
         let mut genesis_path = config.checkpoint_summary_dir.clone();
         genesis_path.push(&config.genesis_filename);
-        Genesis::load(&genesis_path)?.committee()?
+        Genesis::load(&genesis_path)?
+            .committee()
+            .map_err(|e| anyhow!(format!("Cannot load Genesis: {e}")))?
     };
 
+    info!("Extracting effects and events for TID: {}", tid);
     extract_verified_effects_and_events(&full_check_point, &committee, tid)
+        .map_err(|e| anyhow!(format!("Cannot extract effects and events: {e}")))
 }
 
 pub async fn get_verified_object(config: &Config, id: ObjectID) -> anyhow::Result<Object> {
-    let client: Client = Client::new(config.rest_url());
-    let object = client.get_object(id).await?;
+    let iota_client: iota_sdk::IotaClient = IotaClientBuilder::default()
+        .build(config.full_node_url.as_str())
+        .await
+        .unwrap();
```

Shouldn't we return an error instead? There are a few more unwraps in this file in functions that return a Result

---

https://github.com/iotaledger/iota/pull/6066#discussion_r2012133918